WORKSPACE

```diff
@@ -80,92 +42,42 @@ http_archive(
-
-# Load j2cl repository
-http_archive(
-    name = "com_google_j2cl",
-    strip_prefix = "j2cl-master",
-    url = "https://github.com/google/j2cl/archive/master.zip",
-)
-
-#http_archive(
-#    name = "google_bazel_common",
-#    strip_prefix = "bazel-common-1c225e62390566a9e88916471948ddd56e5f111c",
-#    urls = ["https://github.com/google/bazel-common/archive/1c225e62390566a9e88916471948ddd56e5f111c.zip"],
-#)
-
-load("@com_google_j2cl//build_defs:repository.bzl", "load_j2cl_repo_deps")
-
-load_j2cl_repo_deps()
+kotlin_repositories()
 
-load("@com_google_j2cl//build_defs:rules.bzl", "setup_j2cl_workspace")
+kt_register_toolchains()
 
-setup_j2cl_workspace()
+# Java deps from Maven.
 
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-
-_JSINTEROP_BASE_VERSION = "master"
+RULES_JVM_EXTERNAL_TAG = "2.8"
 
-http_archive(
-    name = "com_google_jsinterop_base",
-    strip_prefix = "jsinterop-base-%s" % _JSINTEROP_BASE_VERSION,
-    url = "https://github.com/google/jsinterop-base/archive/%s.zip" % _JSINTEROP_BASE_VERSION,
-)
+RULES_JVM_EXTERNAL_SHA = "79c9850690d7614ecdb72d68394f994fef7534b292c4867ce5e7dec0aa7bdfad"
 
 http_archive(
-    name = "com_google_dagger",
-    urls = ["https://github.com/google/dagger/archive/dagger-2.23.1.zip"],
+    name = "rules_jvm_external",
+    sha256 = RULES_JVM_EXTERNAL_SHA,
+    strip_prefix = "rules_jvm_external-%s" % RULES_JVM_EXTERNAL_TAG,
+    url = "https://github.com/bazelbuild/rules_jvm_external/archive/%s.zip" % RULES_JVM_EXTERNAL_TAG,
 )
 
-maven_jar(
-    name = "com_google_dagger_runtime",
-    artifact = "com.google.dagger:dagger:jar:sources:2.23.1",
-)
-
-maven_jar(
-    name = "javax_inject_source",
-    artifact = "javax.inject:javax.inject:jar:sources:1",
-)
-
-maven_jar(
-    name = "junit",
-    artifact = "junit:junit:4.11",
-)
-
-http_archive(
-    name = "com_google_elemental2",
-    strip_prefix = "elemental2-master",
-    url = "https://github.com/google/elemental2/archive/master.zip",
-)
-
-load("@com_google_elemental2//build_defs:repository.bzl", "load_elemental2_repo_deps")
-
-load_elemental2_repo_deps()
-
-load("@com_google_elemental2//build_defs:workspace.bzl", "setup_elemental2_workspace")
-
-setup_elemental2_workspace()
-
-http_archive(
-    name = "build_bazel_rules_android",
-    sha256 = "cd06d15dd8bb59926e4d65f9003bfc20f9da4b2519985c27e190cddc8b7a7806",
-    strip_prefix = "rules_android-0.1.1",
-    urls = ["https://github.com/bazelbuild/rules_android/archive/v0.1.1.zip"],
-)
+load("@rules_jvm_external//:defs.bzl", "maven_install")
 
 AUTO_VALUE_VERSION = "1.7"
 
-maven_jar(
-    name = "autovalue",
-    artifact = "com.google.auto.value:auto-value:" + AUTO_VALUE_VERSION,
-    sha1 = "fe8387764ed19460eda4f106849c664f51c07121",
-)
-
-maven_jar(
-    name = "autovalue_annotations",
-    artifact = "com.google.auto.value:auto-value-annotations:" + AUTO_VALUE_VERSION,
-    sha1 = "5be124948ebdc7807df68207f35a0f23ce427f29",
+maven_install(
```

Comment:
yay, I was just about to do this (have a version of it in my particle accelerator branch), now I don't have to split it out into separate PRs :)